### PR TITLE
[Upgrade] Bring back secret in prod upgrade

### PR DESCRIPTION
**Pull request by avano**
//skip-ci

### CODE REVIEW

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

File: rest-tests/src/test/resources/upgrade/prod/prod-upgrade.sh [hunk 1]
```diff
@@ -33,23 +33,36 @@ CURRENT_VERSION="$(echo "${CURRENT_BRANCH}" | cut -c1-3)"
...
-#[[ "${CURRENT_VERSION}" != "$(echo "${INSTALL_TAG}" | cut -c1-3)" ]] && echo "Current branch ${CURRENT_BRANCH} does not match install tag ${INSTALL_TAG}" && exit 1
-
+# Create secret for pulling the images
+cat << EOF | oc create -f -
+apiVersion: v1
+kind: Secret
+metadata:
+  name: syndesis-pull-secret
+data:
+  .dockerconfigjson: "${PULL_SECRET}"
```
Comment (asmigala) [inline rest-tests/src/test/resources/upgrade/prod/prod-upgrade.sh]: Shouldn't `$PULL_SECRET` be also mentioned in the doc section above?

File: rest-tests/src/test/resources/upgrade/prod/prod-upgrade.sh [hunk 2]
```diff
@@ -33,23 +33,36 @@ CURRENT_VERSION="$(echo "${CURRENT_BRANCH}" | cut -c1-3)"
...
+EOF
+
+# Install previous version using the scripts from fuse-online-install
 cd "${INSTALL_DIR}" && git checkout "${INSTALL_TAG}"
 bash ./install_ocp.sh --setup
 bash ./install_ocp.sh
 
 echo "Waiting for syndesis deployment..."
 
+# Wait until all insteresting pods are ready
```
Comment (asmigala) [inline rest-tests/src/test/resources/upgrade/prod/prod-upgrade.sh]: typo here

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by avano** [comment]
thx, both fixed

**Pull request merged**